### PR TITLE
Small refactoring block-guard service

### PR DESCRIPTION
**Pull request by Amuhar**

### CODE REVIEW

File: src/guardian/guardian.service.spec.ts [hunk 1]
```diff
@@ -104,7 +103,7 @@ describe('GuardianService', () => {
     }));
 
     const getBlockGuardServiceMock = jest
-      .spyOn(blockGuardService, 'isNeedToProcessNewState')
+      .spyOn(guardianService, 'isNeedToProcessNewState')
```
Comment (eddort) [inline src/guardian/guardian.service.spec.ts:106]: 'blockGuardService' is defined but never used

File: src/guardian/guardian.service.ts [hunk 2]
```diff
@@ -412,4 +412,32 @@ export class GuardianService implements OnModuleInit {
       stakingModuleData.isModuleDepositsPaused
     );
   }
+
+  public isNeedToProcessNewState(newMeta: {
```
Comment (eddort) [inline src/guardian/guardian.service.ts:416]: Could you tell me why you decided to move this method?

File: src/guardian/guardian.service.spec.ts (hunk 1 above)
Comment (Amuhar) [reply src/guardian/guardian.service.spec.ts:106]: changed in fix/val-1216-gurdian-service-impr

File: src/guardian/guardian.service.ts (hunk 2 above)
Comment (Amuhar) [reply src/guardian/guardian.service.ts:416]: https://github.com/lidofinance/lido-council-daemon/pull/220/files#r1732405735 

this cache belong more to guardian.service

Comment (eddort) [reply src/guardian/guardian.service.ts:416]: I have already moved this method, it was here before, I think it is more convenient when there are no auxiliary methods in the main entity. This method looks good in the Blocks service as well

Comment (Amuhar) [reply src/guardian/guardian.service.ts:416]: With this method, we check whether we have already verified the keys on this block, whether we need to process a new state, sign deposit messages, and so on. BlockGuardService does not handle these tasks. Instead, BlockGuardService (which will be renamed to BlockDataCollector in the future) contains methods to fetch and collect data from contracts. It also doesn't know anything about the keys API.

GuardianService is the main module responsible for running methods to check keys on this block, process a new state, sign deposit messages, and so on. It should maintain the state necessary to determine whether we need to process a new state.

**Review by eddort — APPROVED** [review]

**Review by infloop — APPROVED** [review]

**Pull request merged**